### DISCUSSION TITLE
General discussion

### DISCUSSION THREAD

**Discussion by adambuczek** [erap320/LenovoWatchMOD#24] [General]
Just got my watch and I am trying your mod.
Is there any way to make the watch vibrate on EVERY notification? I know next to nothing about android app development but I am eager to fork an learn if it can be done reasonably easily.

**Comment by erap320** [comment]
I'm not an android app developer myself, so I just did substitutions in what I was able to reverse-engineer by decompiling the app.
I would like to make notifications for "every app" possible too, so I'll try to achieve it and let you know if anything comes up :)

If you want to try playing with the apk too, you can use _APK Easy Tool_ like I did.
Sorry but I'm not really comfortable uploading the decompiled code.

**Comment by adambuczek** [comment]
I see. There is also a device request for this watch on gadgetbridge.
https://github.com/Freeyourgadget/Gadgetbridge/issues/1156

**Comment by f3z** [comment]
Hi everyone, just received my device too. Will try and add this to Gadgetbridge and see how I go. Will let you guys know of any progress.

**Comment by f3z** [comment]
I have forked it on my account and will be uploading any updates on there. If any of you want to test it out, let me know.

**Comment by adambuczek** [comment]
@f3z 
![image](https://user-images.githubusercontent.com/3692560/42936296-08340d18-8b4c-11e8-9a53-615bd326bfbc.png)

**Comment by joelbars** [comment]
I volunteer too!! @ERap320, I know that this is more work for you but I wanted to help in some way. I decompiled the file with APKEasyTool and I wanted to know if you could give some directions on how did you added/changed the notifications on Readme.md, is something I was trying to do too. Thanks in advance.

**Comment by erap320** [comment]
As I said about not uploading decompiled code, I also would like not to make this kind of stuff the main point of the repo, but I am happy to tell you about the changes I made.

It mainly was swapping package names of chinese apps with the ones I wanted, as you can see in [Substitutions.txt](https://github.com/ERap320/LenovoWatchMOD/blob/master/Substitutions.txt). This happened in `smali_classes2\com\pmpd\business\device\notify\NotificationService.smali` and `smali_classes2\com\pmpd\business\device\notify\ConstantPackageName.smali`.

The changes to the visible names of the apps were in `res\values\strings.xml`, the icons in `res\mipmap-xhdpi` and `res\mipmap-xxhdpi`

**Comment by joelbars** [comment]
Hi @ERap320, don't worry I understand the reasons for you not put code here and I totally agree with it. These directions will help us to help each other, let's see what we can do and even help Gadgetbridge project. Thanks.

**Comment by f3z** [comment]
There's a pull request for this now: [Freeyourgadget/Gadgetbridge#1156](https://github.com/Freeyourgadget/Gadgetbridge/issues/1156)

**Comment by mfj78** [comment]
Great app mod! It seems the watch vibrates randomly. I have a few persistent notifications. Could that be the issue?
Also the watch vibrates every time I enter and exit the FB Messenger app.

**Comment by erap320** [comment]
@mfj78 I think the watch doesn't like persistent notifications. The same happened to me while listening to WhatsApp voice messages, which for some reason mantain a notification while playing.

I can't say much for the FB app, maybe someone else can help you with that.

**Comment by mfj78** [comment]
> @mfj78 I think the watch doesn't like persistent notifications. The same happened to me while listening to WhatsApp voice messages, which for some reason mantain a notification while playing.
> 
> I can't say much for the FB app, maybe someone else can help you with that.

I see. Just have to live with it then. Thanks man!

**Comment by luissmo** [comment]
Hi!
I'm only here to say thanks to "your" app, I can recibe my most important notifications :) Good work!
Regards!!!

**Comment by DG68BB** [comment]
Hey, I'd like your help regarding the Lenovo Watch App. I've made an account on lenovo.com and when I try to sign in on the app it tells me I need to "bind my email", so I put my email in it asks for a verification that was supposed to be sent out. The problem is I don't receive any email! I've tried numerous times and it doesn't work and so I'd like your help

**Comment by DG68BB** [comment]
I know this has nothing to do with the mod, but I was interested of anyone could help with this... Thank you!

**Comment by eboyid** [comment]
Hi @DG68BB have you solve your problem. Cause I had a same problem here, hhe
